### ISSUE TITLE
Support for map projection

### ISSUE THREAD

**Issue by lrowe** [RedisGraph/RedisGraph#1065]
Once maps are supported I would like to request support for map projection as this would enable efficient construction of GraphQL like responses.

**Comment by DvirDukhan** [comment]
@lrowe you mean returning maps in the resultset, as a `map` datatype?

**Comment by lrowe** [comment]
@DvirDukhan yes, the neo4J docs have a good example:

```
MATCH (actor:Person { name: 'Charlie Sheen' })-[:ACTED_IN]->(movie:Movie)
RETURN actor { .name, .realName, movies: collect(movie { .title, .year })}
```

https://neo4j.com/docs/cypher-manual/current/syntax/maps/#cypher-map-projection

**Comment by Sceat** [comment]
I would definitely use this feature!

**Comment by Sceat** [comment]
was this closed by #1514 ?

**Comment by jeffreylovitz** [comment]
@Sceat It was, yes! We've documented support for the feature (which I believe is total) [here](https://oss.redis.com/redisgraph/datatypes/#map-projections).

Closing this issue now.
